pkg/planner/core/planbuilder.go

```diff
@@ -1309,6 +1338,12 @@ func getPossibleAccessPaths(ctx base.PlanContext, tableHints *hint.PlanHints, in
 			if path := getTablePath(publicPaths); path != nil {
 				hasUseOrForce = true
 				path.Forced = true
+				if i >= indexHintsLen {
```

Comment:
Better to have some checks or test case to ensure the implicit constaint `path.IsIndexLookUpPushDown = indexLookUpPushDown[i]` is hold, and add some comments here too.

Reply:
I've added some comments here. But the tests may be added in another PR because in the current PR the hint is disabled and cannot be tested in integration MySQL tests.

The tests may like this: https://github.com/lcwangchao/tidb/blob/executor_indexlookup_pushdown/tests/integrationtest/t/executor/index_lookup_pushdown.test